### PR TITLE
prepare kahlan v3

### PR DESCRIPTION
**Pull request by samsonasik**
based on https://github.com/crysalead/kahlan/pull/155

### CODE REVIEW

File: spec/FooSpec.php [hunk 1]
```diff
@@ -39,9 +39,9 @@
             $param = 'foo';
             $expected = $param.' processed';
 
-            Stub::on($this->dependency)->method('process')
-                                       ->with($param)
-                                       ->andReturn($expected);
+            allow($this->dependency)->method('process')
```
Comment (jails) [inline spec/FooSpec.php]: Hi, the API is `allow(...)->toReceive(...)` similar to `expect(...)->toReceive(...)`

Comment (samsonasik) [reply spec/FooSpec.php]: with that, I got:

```
Call to undefined method Kahlan\\Matcher\\ToReceive::andReturn()"
```

File: composer.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     },
     "repositories" : [ {
         "type" : "vcs",
-        "url" : "https://github.com/crysalead/kahlan.git"
+        "url" : "https://github.com/jails/kahlan.git"
```
Comment (jails) [inline composer.json]: The repo is `crysalead/kahlan` but will be transfered to `kahlan/kahlan` for the 3.0. I'll keep you posted ;-)

Comment (samsonasik) [reply composer.json]: I updated.

**Comment by samsonasik** [comment]
I was using `allow()`, but got:

```
PHP Fatal error:  Call to undefined method Kahlan\Allow::method()
 in /home/travis/build/sitrunlab/learn-kahlan/spec/FooSpec.php on line 42
```

it seems something issue with function autoloading ?

**Comment by jails** [comment]
It's correct to use `allow()` to create a stub. But it's `allow(...)->toReceive(..)` not `allow(...)->method(...)`.

**Comment by samsonasik** [comment]
it seems working now, thank you, running locally: 

```
Apples-MacBook-Pro:learn-kahlan samsonasik$ bin/kahlan
            _     _
  /\ /\__ _| |__ | | __ _ _ __
 / //_/ _` | '_ \| |/ _` | '_ \
/ __ \ (_| | | | | | (_| | | | |
\/  \/\__,_|_| |_|_|\__,_|_| |_|

The Unit/BDD PHP Test Framework for Freedom, Truth, and Justice.

Working Directory: /Users/samsonasik/www/learn-kahlan

......                                                              6 / 6 (100%)



Expectations   : 6 Executed
Specifications : 0 Pending, 0 Excluded, 0 Skipped

Passed 6 of 6 PASS in 0.106 seconds (using 2Mo)
```

travis seems need waiting

**Comment by jails** [comment]
:+1:

**Comment by samsonasik** [comment]
merged with kahlan/kahlan:^3.0.0, thanks @jails ;)

**Comment by jails** [comment]
Awesome !!

**Pull request merged**